gradle-android-command-plugin/src/main/groovy/com/novoda/gradle/command/AdbTask.groovy

```diff
@@ -70,7 +110,7 @@ public class AdbTask extends org.gradle.api.DefaultTask {
             it.startsWith("$propertyKey:")
         }
         if (output == null) {
-            throw new IllegalStateException("Could not read property '$propertyKey' of $apkPath")
+            throw new IllegalArgumentException("Could not read property '$propertyKey' of $apkPath")
```

Comment:
Two reasons: First, I need the caller to be able to distinguish the `output == null` case from the other IllegalStateException thrown in line 107 (of the new code). Second, IMO an IllegalStateException is not semantically correct in this case. If `output == null` it does not mean the plugin is in a wrong state, but that there was no match for the `propertyKey` argument, hence an IllegalArgumentException seems more appropriate to me.